### ISSUE TITLE
Add grouping functionality

### ISSUE THREAD

**Issue by zenalapp** [SchlossLab/mikRopML#100]
Data is split into train/test sets where a group is only included in the training set or test set.

- [x] Implement code
- [x] Documentation in code (#104 for vignette issue)
